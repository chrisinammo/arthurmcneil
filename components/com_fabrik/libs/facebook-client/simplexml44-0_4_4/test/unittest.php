<?php

// $Id: unittest.php,v 1.4 2005/01/20 20:19:43 ingo Exp $

ini_set('include_path', './:../class:../../lib:./PEAR');

include_once 'PHPUnit.php';
include_once 'test.setup';

// APD
if( function_exists('apd_set_pprof_trace') ) {
  apd_set_pprof_trace(); }

ini_set("error_reporting", E_ALL);

$testcount  = 0;
$testpassed = 0;
$testfailed = 0;

function print_header($string)
{
  $str  = "\nrunning regression tests on ".$string;
  $str .= "\n".(underline($str))."\n";
  print $str;
}

function underline($str)
{
  $line = '';
  for ( $i = 1; $i < strlen($str); $i++)
    $line .= '-';
  return $line;
}

function versionTest()
{
  global $minversion;
  $version = phpversion();
  $expectstr = "expected at least PHP $minversion";

  print "\n";
  if( version_compare($minversion, $version) > 0 )
    {
      print "WARNING: found PHP $version, $expectstr\n";
    }
  else
    {
      print "OK, found PHP $version, $expectstr\n";
    }
}

versionTest();

$dir = opendir('./');

while( $file = readdir($dir) )
  {
    if (preg_match("/^$prefix/", $file))
      $tests[] = basename( $file, ".php");
  }

if(! count($tests) )
{
  print "Hm, no tests found...\n";
  exit(1);
}

sort($tests);

foreach( $tests as $test )
    {
      include "$test".".php";
      
      print_header($test);

      $suite  = new PHPUnit_TestSuite($test);
      $result = PHPUnit::run($suite);
      $count  = $result->runCount();
      $pass   = count($result->passedTests());
      $fail   = $count - $pass;
      $testcount  += $count;
      $testpassed += $pass;
      $testfailed += $fail;
      print $result->toString();
      
    }

print "\n";
print "Summary\n".(underline('Summary '))."\n";
print "executed $testcount tests: $testpassed passed, $testfailed failed\n\n";

?>
