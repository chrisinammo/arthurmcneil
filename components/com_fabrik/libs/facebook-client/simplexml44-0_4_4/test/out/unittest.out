
OK, found PHP 4.3.10, expected at least PHP 4.2.0

running regression tests on TestXml01IsterXmlNode
-------------------------------------------------
TestCase testxml01isterxmlnode->testappend() passed
TestCase testxml01isterxmlnode->testasxml() passed
TestCase testxml01isterxmlnode->testgetroot() passed
TestCase testxml01isterxmlnode->testhaschildren() passed
TestCase testxml01isterxmlnode->testxpath() passed
TestCase testxml01isterxmlnode->testtosimplexml() passed
TestCase testxml01isterxmlnode->testtodom() passed

running regression tests on TestXml02IsterXmlExpat
--------------------------------------------------
TestCase testxml02isterxmlexpat->testsetsourcefile() passed
TestCase testxml02isterxmlexpat->testsetsourcestring() passed
TestCase testxml02isterxmlexpat->testparsefile() passed
TestCase testxml02isterxmlexpat->testparsestring() passed
TestCase testxml02isterxmlexpat->testlocate() passed
TestCase testxml02isterxmlexpat->testgetsetparser() passed

running regression tests on TestXml03IsterXmlExpatDumper
--------------------------------------------------------
TestCase testxml03isterxmlexpatdumper->testparse() passed

running regression tests on TestXml04IsterXmlExpatNonValid
----------------------------------------------------------
TestCase testxml04isterxmlexpatnonvalid->testparse() passed
TestCase testxml04isterxmlexpatnonvalid->testentitydeclaration() passed
TestCase testxml04isterxmlexpatnonvalid->testnotationdeclaration() passed
TestCase testxml04isterxmlexpatnonvalid->testtag() passed
TestCase testxml04isterxmlexpatnonvalid->testattributes() passed
TestCase testxml04isterxmlexpatnonvalid->testcdatasection() passed
TestCase testxml04isterxmlexpatnonvalid->testcomment() passed
TestCase testxml04isterxmlexpatnonvalid->testpiparse() passed
TestCase testxml04isterxmlexpatnonvalid->testpirun() passed
TestCase testxml04isterxmlexpatnonvalid->testparsedentity() passed
TestCase testxml04isterxmlexpatnonvalid->testexternalentity() passed
TestCase testxml04isterxmlexpatnonvalid->testnamespace() passed
TestCase testxml04isterxmlexpatnonvalid->teststringbug1() passed
TestCase testxml04isterxmlexpatnonvalid->teststringbug2() passed
TestCase testxml04isterxmlexpatnonvalid->testunicode() passed

running regression tests on TestXml05IsterSimpleXMLElement
----------------------------------------------------------
TestCase testxml05istersimplexmlelement->testaccessnode() passed
TestCase testxml05istersimplexmlelement->testchildren1() passed
TestCase testxml05istersimplexmlelement->testchildren2() passed
TestCase testxml05istersimplexmlelement->testchildren3() passed
TestCase testxml05istersimplexmlelement->testchildren4() passed
TestCase testxml05istersimplexmlelement->testaccesscdata() passed
TestCase testxml05istersimplexmlelement->testaccessattributes() passed
TestCase testxml05istersimplexmlelement->testaccessparsedentity() passed
TestCase testxml05istersimplexmlelement->testaccessexternalentity() passed
TestCase testxml05istersimplexmlelement->testsetcdata() passed
TestCase testxml05istersimplexmlelement->testsetattribute() passed

running regression tests on TestXml06IsterXmlSimpleXMLImpl
----------------------------------------------------------
TestCase testxml06isterxmlsimplexmlimpl->testload_file() passed
TestCase testxml06isterxmlsimplexmlimpl->testload_string() passed
TestCase testxml06isterxmlsimplexmlimpl->testsetcdata1() passed
TestCase testxml06isterxmlsimplexmlimpl->testsetcdata2() passed
TestCase testxml06isterxmlsimplexmlimpl->testsetattribute1() passed
TestCase testxml06isterxmlsimplexmlimpl->testsetattribute2() passed

Summary
-------
executed 46 tests: 46 passed, 0 failed

